### PR TITLE
Initial null service

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-4173

## Changes
- Add NullService and support for creating null service object
- Fix Portugal postal code

### CODE REVIEW

File: src/foam/core/NullServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package foam.core;
```
Comment (kgrgreer) [inline src/foam/core/NullServiceTest.java:1]: Copyright.

File: src/foam/core/NullService.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * NullService facilitates the creation of null object of a given interface.
+ *
+ * Thus removing the need to manually create class for the null implementation
+ * of the service interface eg. NullMyService.
+ *
+ * See. NullServiceTest.java for usage.
+ *
+ */
+public class NullService {
```
Comment (kgrgreer) [inline src/foam/core/NullService.java:22]: How often will this be useful? Even for Null services don't you need to know what the null operation is? For example, a validator interface would still need to know if the default NOP was to return true or false.

File: src/foam/core/NullServiceTest.java (hunk 1 above)
Comment (lchanmann) [reply src/foam/core/NullServiceTest.java:7]: Fixed in https://github.com/kgrgreer/foam3/pull/101/commits/d41fd31a0ae40be99c125b3e4345c3d67a4f7d0f.

File: src/foam/core/NullService.java (hunk 2 above)
Comment (kgrgreer) [reply src/foam/core/NullService.java:22]: If the interface is implemented with default operations like these, then you could do the same by just implementing the interface (and extending more easily where needed).

Comment (lchanmann) [reply src/foam/core/NullService.java:22]: This is at runtime, we can spin off a null service on the fly without the need to add code, re-compile, re-build and re-deploy.

Might not be useful for a null service that we want to control what the output is. Can be handy for creating null service for interfaces having more void (no return value) methods.

**Pull request closed without merging**